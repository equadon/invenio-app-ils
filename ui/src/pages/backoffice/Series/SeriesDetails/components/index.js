export { SeriesLiterature } from './SeriesLiterature';
export { SeriesMetadata } from './SeriesMetadata';
export { SeriesRelations } from './SeriesRelations';
