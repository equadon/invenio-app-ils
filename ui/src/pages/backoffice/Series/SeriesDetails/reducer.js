export { default as seriesDetailsReducer } from './state/reducer';
export {
  default as seriesLiteratureReducer,
} from './components/SeriesLiterature/state/reducer';
export {
  default as seriesRelationsReducer,
} from './components/SeriesRelations/state/reducer';
