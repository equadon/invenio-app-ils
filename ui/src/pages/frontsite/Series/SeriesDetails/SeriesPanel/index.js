import { connect } from 'react-redux';
import SeriesPanelComponent from './SeriesPanel';
import SeriesPanelMobileComponent from './SeriesPanelMobile';

const mapStateToProps = state => ({
  series: state.seriesDetailsFront.data,
  isLoading: state.seriesDetailsFront.isLoading,
});

export const SeriesPanel = connect(
  mapStateToProps,
  null
)(SeriesPanelComponent);

export const SeriesPanelMobile = connect(
  mapStateToProps,
  null
)(SeriesPanelMobileComponent);
