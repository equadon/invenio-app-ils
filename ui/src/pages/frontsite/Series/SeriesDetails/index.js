import { connect } from 'react-redux';
import { fetchSeriesDetails } from './state/actions';
import SeriesDetailsContainerComponent from './SeriesDetails';

const mapDispatchToProps = dispatch => ({
  fetchSeriesDetails: pid => dispatch(fetchSeriesDetails(pid)),
});

const mapStateToProps = state => ({
  isLoading: state.seriesDetailsFront.isLoading,
  series: state.seriesDetailsFront.data,
  hasError: state.seriesDetailsFront.hasError,
});

export const SeriesDetailsContainer = connect(
  mapStateToProps,
  mapDispatchToProps
)(SeriesDetailsContainerComponent);
