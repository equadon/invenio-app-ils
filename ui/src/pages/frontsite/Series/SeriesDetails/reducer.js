export { default as seriesDetailsFrontReducer } from './state/reducer';
