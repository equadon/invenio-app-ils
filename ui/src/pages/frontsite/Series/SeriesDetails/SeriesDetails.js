import React from 'react';
import PropTypes from 'prop-types';
import { Container } from 'semantic-ui-react';
import { Error, SearchBar } from '@components';
import { goTo } from '@history';
import { FrontSiteRoutes } from '@routes/urls';
import { ILSParagraphPlaceholder } from '@components/ILSPlaceholder';
import { Breadcrumbs } from '@pages/frontsite/components';
import { SeriesPanel } from './SeriesPanel';
import { SeriesLiterature } from '@pages/frontsite/components/Series';
import { SeriesMetadata } from './SeriesMetadata';

export default class SeriesDetails extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      searchQuery: '',
    };
    this.anchors = {
      literature: React.createRef(),
    };
  }

  componentDidMount() {
    this.unlisten = this.props.history.listen(location => {
      if (location.state && location.state.seriesPid) {
        this.props.fetchSeriesDetails(location.state.seriesPid);
      }
    });
    this.props.fetchSeriesDetails(this.props.match.params.seriesPid);
  }

  componentWillUnmount() {
    this.unlisten();
  }

  breadcrumbs = () => [
    { to: FrontSiteRoutes.home, label: 'Home' },
    { to: FrontSiteRoutes.documentsList, label: 'Search' },
  ];

  onSearchClick = () => {
    const query = encodeURIComponent(this.state.searchQuery);
    goTo(FrontSiteRoutes.documentsListWithQuery(query));
  };

  onSearchInputChange = (value, event) => {
    this.setState({ searchQuery: event.target.value });
  };

  render() {
    const { error, isLoading, series } = this.props;
    return (
      <>
        <Container fluid className="document-details-search-container">
          <Container>
            <SearchBar
              currentQueryString={this.state.searchQuery}
              onInputChange={this.onSearchInputChange}
              executeSearch={this.onSearchClick}
              placeholder="Search for books..."
              className="document-details-search-bar"
            />
          </Container>
        </Container>
        <Error boundary error={error}>
          <Container className="document-details-container default-margin-top">
            <ILSParagraphPlaceholder isLoading={isLoading} lines={1}>
              <Breadcrumbs
                isLoading={isLoading}
                elements={this.breadcrumbs()}
                currentElement={series.metadata ? series.metadata.title : null}
              />
            </ILSParagraphPlaceholder>
            <SeriesPanel anchors={this.anchors} />
          </Container>
          <Container className="items-locations spaced">
            <ILSParagraphPlaceholder linesNumber={3} isLoading={isLoading}>
              <div ref={this.anchors.literature}>
                <SeriesLiterature />
              </div>
            </ILSParagraphPlaceholder>
          </Container>
          <Container className="section" fluid>
            <Container>
              <ILSParagraphPlaceholder linesNumber={20} isLoading={isLoading}>
                <SeriesMetadata />
              </ILSParagraphPlaceholder>
            </Container>
          </Container>
        </Error>
      </>
    );
  }
}

SeriesDetails.propTypes = {
  series: PropTypes.object.isRequired,
};
