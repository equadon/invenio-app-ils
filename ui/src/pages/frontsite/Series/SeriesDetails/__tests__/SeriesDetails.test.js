import React from 'react';
import { shallow, mount } from 'enzyme';
import testData from '@testData/series.json';

import SeriesDetails from '../SeriesDetails';
import { BrowserRouter } from 'react-router-dom';

jest.mock('../SeriesPanel', () => {
  return {
    SeriesPanel: () => null,
  };
});
jest.mock('../SeriesMetadata', () => {
  return {
    SeriesMetadata: () => null,
  };
});

jest.mock('@pages/frontsite/components/Series', () => {
  return {
    SeriesTags: () => null,
  };
});

describe('SeriesDetailsContainer tests', () => {
  let component;
  afterEach(() => {
    if (component) {
      component.unmount();
    }
  });

  const routerHistory = {
    listen: () => () => {},
  };
  const routerUrlParams = {
    params: {
      seriesPid: 111,
    },
  };

  it('should load the details component', () => {
    const component = shallow(
      <SeriesDetails
        history={routerHistory}
        match={routerUrlParams}
        fetchSeriesDetails={() => {}}
        series={{ metadata: testData[0] }}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should fetch details on mount', () => {
    const mockedFetchSeriesDetails = jest.fn();
    component = mount(
      <BrowserRouter>
        <SeriesDetails
          history={routerHistory}
          match={routerUrlParams}
          fetchSeriesDetails={mockedFetchSeriesDetails}
          series={{ metadata: testData[0] }}
        />
      </BrowserRouter>
    );
    expect(mockedFetchSeriesDetails).toHaveBeenCalledWith(
      routerUrlParams.params.seriesPid
    );
  });
});
