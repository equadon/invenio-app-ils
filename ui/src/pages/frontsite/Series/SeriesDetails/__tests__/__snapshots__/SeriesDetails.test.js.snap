// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeriesDetailsContainer tests should load the details component 1`] = `
<Fragment>
  <Container
    className="document-details-search-container"
    fluid={true}
  >
    <Container>
      <SearchBar
        className="series-details-search-bar"
        currentQueryString=""
        executeSearch={[Function]}
        onInputChange={[Function]}
        placeholder="Search for books..."
      />
    </Container>
  </Container>
  <Error
    boundary={true}
  >
    <Container
      className="series-details-container default-margin-top"
    >
      <ILSParagraphPlaceholder
        lines={1}
      >
        <Breadcrumbs
          currentElement="The Gulf: The Making of An American Sea"
          elements={
            Array [
              Object {
                "label": "Home",
                "to": "/",
              },
              Object {
                "label": "Search",
                "to": "/search",
              },
            ]
          }
        />
      </ILSParagraphPlaceholder>
      <SeriesPanel />
    </Container>
    <Container
      className="items-locations spaced"
    >
      <ILSParagraphPlaceholder
        linesNumber={3}
      >
        <Component />
      </ILSParagraphPlaceholder>
    </Container>
    <Container
      className="section"
      fluid={true}
    >
      <Container>
        <ILSParagraphPlaceholder
          linesNumber={20}
        >
          <SeriesMetadata />
        </ILSParagraphPlaceholder>
      </Container>
    </Container>
  </Error>
</Fragment>
`;
